pyomo/core/base/misc.py

```diff
@@ -118,7 +118,7 @@ def __init__(self, key=None):
         self._typemap[tuple] =(3, tuple.__name__)
         self._key = key
 
-    def __call__(self, val):
+    def __call__(self, val, use_key=True):
```

Comment:
Instead of changing the signature for `__call__`, I think I would prefer adding a helper function:
```python
def __call__(self, val):
    #...
    if self._key is not None:
        val = self._key(val)
    return self._generate_key(val)

def _generate_key(self, val):
    # the rest of what used to be `__call__`
    #...
    elif i == 3:
        return _typename, tuple(self._generate_key(v) for v in val)
    #...
```